packages/react-native/React/Views/UIView+React.h

```diff
@@ -8,11 +8,17 @@
 #import <UIKit/UIKit.h>
 
 #import <React/RCTComponent.h>
-#import <React/RCTTransformOrigin.h>
+#import <React/RCTConvert.h>
```

Comment:
Would it be possible to factor this in a way where it didn't depends on `RCTConvert.h` in the header? This brings in a bunch of other bits like WebKit headers, which makes the build unhappy of some modules which import this file.

Reply:
Hmm, looking at `RCTConvert.h`, it only wants the header for the type `WKDetectorTypes`, of which there is an equivalent `UIDetectorTypes`, that is.. also defined and used. How about I just simplify the logic and only use `UIDetectorTypes`? Ideally.. We shouldn't be depending on Webkit at all, given Hermes exists...

Reply:
If we like that, I can spin up a separate PR, land that, then revisit this one.

Reply:
That sounds great!

Reply:
https://github.com/facebook/react-native/pull/39794/
Should build...

Reply:
Neat, I can probably extend this to macOS too lol https://developer.apple.com/documentation/foundation/nsdatadetector

Reply:
er, maybe this one? I dunno blurgh https://developer.apple.com/documentation/appkit/nstextinputtraits/3242733-datadetectiontype